### PR TITLE
Enforce cargo fmt

### LINKED ISSUE #198: enforce cargo fmt for PRs
We should require all PRs to use rustfmt.

### PR DESCRIPTION
**Pull request by tmandry**
Closes #198.

### CODE REVIEW

**Comment by tmandry** [comment]
Feel free to merge some things in ahead of this and I'll rebase.. it'll be easier that way.

**Comment by nikomatsakis** [comment]
Travis seems grumpy. One thing:

```
$ RUSTC_BOOTSTRAP=1 cargo fmt --all -- --check
error: 'cargo-fmt' is not installed for the toolchain 'nightly-x86_64-unknown-linux-gnu'
To install, run `rustup component add rustfmt`
The command "RUSTC_BOOTSTRAP=1 cargo fmt --all -- --check" exited with 1.
```

Other things:

```
error[E0432]: unresolved import `ena::unify`
 --> chalk-solve/src/infer.rs:3:5
  |
3 | use ena::unify as ena;
  |     ^^^^^^^^^^^^^^^^^ no `unify` in `unify`
error[E0432]: unresolved import `ena::unify`
 --> chalk-solve/src/infer.rs:3:5
  |
3 | use ena::unify as ena;
  |     ^^^^^^^^^^^^^^^^^ no `unify` in `unify`
```

although those things might just be bugs that were fixed

File: chalk-solve/src/infer.rs [hunk 1]
```diff
@@ -1,15 +1,15 @@
-use ::ena::unify as ena;
-use chalk_ir::*;
 use chalk_ir::fold::Fold;
+use chalk_ir::*;
+use ena::unify as ena;
```
Comment (bjorn3) [inline chalk-solve/src/infer.rs]: The leading `::` got removed.

**Comment by tmandry** [comment]
Oops, my nightly was outdated. Trying again.

I also enable Travis caching of cargo in this PR. I don't anticipate any problems with this but if you know of any I'll take it out.

File: .travis.yml [hunk 2]
```diff
@@ -2,9 +2,13 @@ language: rust
 rust:
   - beta
   - nightly
+cache: cargo
```
[the remark this answers is not in the document]
Comment (nikomatsakis) [reply .travis.yml]: Do we care ?

[the remark this answers is not in the document]
Comment (nikomatsakis) [reply .travis.yml]: That is, I don't really know what the implications of that are.

**Pull request merged**